addons/mail/models/discuss/discuss_channel.py

```diff
@@ -405,14 +405,15 @@ def add_members(self, partner_ids=None, guest_ids=None, invite_to_rtc_call=False
                     else:
                         notification = (Markup('<div class="o_mail_notification">%s</div>') % _("invited %s to the channel")) % member.partner_id._get_html_link()
                     member.channel_id.message_post(body=notification, message_type="notification", subtype_xmlid="mail.mt_comment")
-            for member in new_members.filtered(lambda member: member.guest_id):
-                member.channel_id.message_post(body=Markup('<div class="o_mail_notification">%s</div>') % _('joined the channel'),
-                    message_type="notification", subtype_xmlid="mail.mt_comment")
-                guest = member.guest_id
-                if guest:
-                    notifications.append((guest, 'discuss.channel/joined', {
-                        'channel': member.channel_id.sudo()._channel_info()[0],
-                    }))
+            if post_joined_message:
+                for member in new_members.filtered(lambda member: member.guest_id):
+                    member.channel_id.message_post(body=Markup('<div class="o_mail_notification">%s</div>') % _('joined the channel'),
+                        message_type="notification", subtype_xmlid="mail.mt_comment")
+                    guest = member.guest_id
+                    if guest:
+                        notifications.append((guest, 'discuss.channel/joined', {
+                            'channel': member.channel_id.sudo()._channel_info()[0],
+                        }))
```

Comment:
I don't think join bus notif should be conditioned to sending the joined message